### PR TITLE
fix: allow ESLint 9 as peer dependency

### PR DESCRIPTION
**Pull request by SimenB**

### CODE REVIEW

**Comment by G-Rath** [comment]
Beat me to it 😁

File: package.json [hunk 1]
```diff
@@ -86,7 +86,7 @@
     "babel-jest": "^29.0.0",
     "babel-plugin-replace-ts-export-assignment": "^0.0.2",
     "dedent": "^1.5.0",
-    "eslint": "^7.0.0 || ^8.0.0",
+    "eslint": "^7.0.0 || ^8.0.0 || ^9.0.0",
```
Comment (G-Rath) [inline package.json:89]: @SimenB is there some Yarn trick I don't know about to actually select the version to use locally without a git change?

I'm finding that v9 is being installed, which completely breaks my IDE and tests - alternatively, could we remove v9 from this list? CI was running fine with installing `next` without this so I don't know quite why we need it 😅

**Comment by anthony0030** [comment]
I am getting this warning:
```LOG
warning "eslint-plugin-jest > @typescript-eslint/utils@6.21.0" has incorrect peer dependency "eslint@^7.0.0 || ^8.0.0".
```
Is it related?

**Comment by SimenB** [comment]
@anthony0030 https://github.com/typescript-eslint/typescript-eslint/issues/8211

(we will probably need to update to v7 whenever that is actually fixed)

Things should work fine even with the warning, tho.

**Comment by anthony0030** [comment]
@SimenB thankyou for the info 😃

File: package.json (hunk 1 above)
Comment (G-Rath) [reply package.json:89]: Have opened https://github.com/jest-community/eslint-plugin-jest/pull/1569

**Pull request merged**